crates/allocator/src/bump.rs

```diff
@@ -174,9 +174,10 @@ impl InnerAlloc {
     /// Note: This implementation results in internal fragmentation when allocating across
     /// pages.
     fn alloc(&mut self, layout: Layout) -> Option<usize> {
-        let alloc_start = self.next;
+        let alloc_start = self.align_ptr(&layout);
+
+        let aligned_size = layout.size();
 
-        let aligned_size = layout.pad_to_align().size();
```

Comment:
Just out of curiosity, how is padding performed for fields in a struct, considering they field types have their own alignment requirements?

Reply:
That is not really the concern of the allocator. The allocator will just create an allocation that adheres to the `Layout` passed. For a struct the alignment will be set to the alignment for the whole struct (AFAIK the alignment of the largest type in the struct). The padding within the struct is given implicitly by the size of the `Layout`. Rust will just assume the padding when accessing the struct.